### PR TITLE
Migrating tests from mocha to jest

### PR DESCRIPTION
**Pull request by evgeniytar**
+ winston

### CODE REVIEW

File: test/mapper.spec.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import logger from '../src/logger'
 logger.silent = true
 
 describe('Mapper', () => {
-  it('maps input to streams', done => {
```
Comment (alexey-pelykh) [inline test/mapper.spec.ts]: https://stackoverflow.com/questions/45778192/what-is-the-difference-between-it-and-test-in-jest maybe we should just leave "it" for wording purposes?

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  preset: 'ts-jest',
+  clearMocks: true,
+  coverageDirectory: "tests/coverage",
```
Comment (alexey-pelykh) [inline jest.config.js]: `'` should be instead of `"`

File: package.json [hunk 3]
```diff
@@ -29,6 +29,11 @@
   "bugs": {
     "url": "https://github.com/dataplug-io/dataplug/issues"
   },
+  "jest": {
+    "collectCoverage": true,
+    "collectDirectory": "./tests/coverage/lcov.info",
+    "collectCoverageFrom": ["**/*.{js,jsx}"]
```
Comment (alexey-pelykh) [inline package.json]: js vs ts? I'm not sure how to do that properly

File: package.json [hunk 4]
```diff
@@ -39,21 +44,20 @@
...
+    "@types/jest": "^24.0.11",
     "@types/json-schema": "^7.0.3",
     "@types/lodash": "^4.14.123",
     "@types/mocha": "^5.2.6",
     "@types/node": "^11.13.0",
-    "chai": "^4.2.0",
-    "chai-as-promised": "^7.1.1",
     "coveralls": "^3.0.3",
+    "jest": "^24.7.1",
     "mocha": "^6.1.1",
```
Comment (alexey-pelykh) [inline package.json]: mocha is not needed anymore

File: package.json [hunk 5]
```diff
@@ -39,21 +44,20 @@
   },
   "devDependencies": {
     "@types/bluebird": "^3.5.26",
-    "@types/chai": "^4.1.7",
-    "@types/chai-as-promised": "^7.1.0",
     "@types/check-types": "^7.3.1",
+    "@types/jest": "^24.0.11",
     "@types/json-schema": "^7.0.3",
     "@types/lodash": "^4.14.123",
     "@types/mocha": "^5.2.6",
```
Comment (alexey-pelykh) [inline package.json]: mocha is not needed anymore

File: src/logger.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import * as winston from 'winston'
+
+const logger = winston.createLogger({
```
Comment (alexey-pelykh) [inline src/logger.ts]: This is not the best approach. It's a library that should not do any default configuration of the logger.

File: test/counter.spec.ts [hunk 7]
```diff
@@ -1,17 +1,13 @@
...
 
-import 'mocha'
-import chai from 'chai'
-import chaiAsPromised from 'chai-as-promised'
+import 'ts-jest'
 import { Promise as BluebirdPromise } from 'bluebird'
 import { PassThrough } from 'stream'
 import Counter from '../src/counter'
-import * as logger from 'winston'
+import logger from '../src/logger'
```
Comment (alexey-pelykh) [inline test/counter.spec.ts]: for tests use separate logger

File: test/index.spec.ts [hunk 8]
```diff
@@ -1,87 +1,83 @@
...
-import 'mocha'
-import chai from 'chai'
-chai.should()
+import 'ts-jest'
 import { dataplug } from '../src'
 
 describe('dataplug', () => {
   it('has "config" field', () => {
-    dataplug.should.have.property('config').that.is.an('object')
+    expect(typeof dataplug.config).toBe('object')
```
Comment (alexey-pelykh) [inline test/index.spec.ts:10]: first check existence of this property

File: test/mapper.spec.ts [hunk 9]
```diff
@@ -150,7 +154,7 @@ describe('Mapper', () => {
     input.end()
   })
 
-  it('handles errors', done => {
+  test('handles errors', done => {
```
Comment (alexey-pelykh) [inline test/mapper.spec.ts]: lets' keep `it` instead of `test` for wording purposes

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

File: test/counter.spec.ts (hunk 7 above)
Comment (evgeniytar) [reply test/counter.spec.ts]: we don't use logger here at all, it imports here to switch off input in console

Comment (alexey-pelykh) [reply test/counter.spec.ts]: then just import winston and disable it directly or configure in otherwise - within test

**Review by alexey-pelykh — APPROVED** [review]

**Pull request merged**